middlewares/stripPrefixRegex_test.go

```diff
@@ -1,55 +1,99 @@
-		fmt.Fprint(w, r.URL.Path)
-	})
-
-	handler := NewStripPrefixRegex(handlerPath, []string{"/a/api/", "/b/{regex}/", "/c/{category}/{id:[0-9]+}/"})
-	server := httptest.NewServer(handler)
-	defer server.Close()
+	testPrefixRegex := []string{"/a/api/", "/b/{regex}/", "/c/{category}/{id:[0-9]+}/"}
 
 	tests := []struct {
-		expectedCode     int
-		expectedResponse string
-		url              string
+		desc               string
+		path               string
+		expectedStatusCode int
+		expectedPath       string
+		expectedHeader     string
 	}{
-		{url: "/a/test", expectedCode: 404, expectedResponse: "404 page not found\n"},
-		{url: "/a/api/test", expectedCode: 200, expectedResponse: "test"},
-
-		{url: "/b/api/", expectedCode: 200, expectedResponse: ""},
-		{url: "/b/api/test1", expectedCode: 200, expectedResponse: "test1"},
-		{url: "/b/api2/test2", expectedCode: 200, expectedResponse: "test2"},
-
-		{url: "/c/api/123/", expectedCode: 200, expectedResponse: ""},
-		{url: "/c/api/123/test3", expectedCode: 200, expectedResponse: "test3"},
-		{url: "/c/api/abc/test4", expectedCode: 404, expectedResponse: "404 page not found\n"},
+		{
+			desc:               "case 01",
+			path:               "/a/test",
+			expectedStatusCode: 404,
+		},
+		{
+			desc:               "case 02",
+			path:               "/a/api/test",
+			expectedStatusCode: 200,
+			expectedPath:       "test",
+			expectedHeader:     "/a/api/",
+		},
+		{
+			desc:               "case 03",
+			path:               "/b/api/",
+			expectedStatusCode: 200,
+			expectedHeader:     "/b/api/",
+		},
+		{
+			desc:               "case 04",
+			path:               "/b/api/test1",
+			expectedStatusCode: 200,
+			expectedPath:       "test1",
+			expectedHeader:     "/b/api/",
+		},
+		{
+			desc:               "case 05",
+			path:               "/b/api2/test2",
+			expectedStatusCode: 200,
+			expectedPath:       "test2",
+			expectedHeader:     "/b/api2/",
+		},
+		{
+			desc:               "case 06",
+			path:               "/c/api/123/",
+			expectedStatusCode: 200,
+			expectedHeader:     "/c/api/123/",
+		},
+		{
+			desc:               "case 07",
+			path:               "/c/api/123/test3",
+			expectedStatusCode: 200,
+			expectedPath:       "test3",
+			expectedHeader:     "/c/api/123/",
+		},
+		{
+			desc:               "case 08",
+			path:               "/c/api/abc/test4",
+			expectedStatusCode: 404,
+		},
 	}
 
 	for _, test := range tests {
-		resp, err := http.Get(server.URL + test.url)
-		if err != nil {
-			t.Fatal(err)
-		}
-		if resp.StatusCode != test.expectedCode {
-			t.Fatalf("Received non-%d response: %d\n", test.expectedCode, resp.StatusCode)
-		}
-		response, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			t.Fatal(err)
-		}
-		if test.expectedResponse != string(response) {
-			t.Errorf("Expected '%s' :  '%s'\n", test.expectedResponse, response)
-		}
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
```

Comment:
For the test cases at hand, it would also suffice to pass `path` to `t.Run` as it's disambiguating already, and the `desc`riptions aren't overly -- descriptive. :)